server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/PrepareModelRelationshipVisitor.java

```diff
@@ -0,0 +1,51 @@
+package org.opentosca.toscana.plugins.cloudformation.visitor;
+
+import org.opentosca.toscana.model.node.Compute;
+import org.opentosca.toscana.model.node.MysqlDatabase;
+import org.opentosca.toscana.model.node.RootNode;
+import org.opentosca.toscana.model.node.WebApplication;
+import org.opentosca.toscana.model.relation.ConnectsTo;
+import org.opentosca.toscana.model.relation.RootRelationship;
+import org.opentosca.toscana.model.visitor.RelationshipVisitor;
+
+import com.scaleset.cfbuilder.core.Fn;
+import org.jgrapht.Graph;
+import org.slf4j.Logger;
+
+import static org.opentosca.toscana.plugins.cloudformation.CloudFormationLifecycle.toAlphanumerical;
+import static org.opentosca.toscana.plugins.cloudformation.visitor.CloudFormationNodeVisitor.getCompute;
+import static org.opentosca.toscana.plugins.cloudformation.visitor.PrepareModelNodeVisitor.AWS_ENDPOINT_REFERENCE;
+
+public class PrepareModelRelationshipVisitor implements RelationshipVisitor {
+
+    private final Logger logger;
+    private Graph<RootNode, RootRelationship> topology;
+
+    public PrepareModelRelationshipVisitor(Logger logger, Graph<RootNode, RootRelationship> topology) {
+        this.logger = logger;
+        this.topology = topology;
+    }
+
+    @Override
+    public void visit(ConnectsTo relation) {
+        logger.info("Prepare ConnectsTo relation {}.", relation.getEntityName());
+        RootNode source = topology.getEdgeSource(relation);
+        RootNode target = topology.getEdgeTarget(relation);
+        if (source instanceof WebApplication && target instanceof MysqlDatabase) {
+            MysqlDatabase mysqlDatabase = (MysqlDatabase) target;
+            WebApplication webApplication = (WebApplication) source;
+            // if they are hosted on the same compute --> we can set the compute private address to
+            Compute computeMysqlDatabase = getCompute(mysqlDatabase);
+            Compute computeWebApplication = getCompute(webApplication);
+            if (computeMysqlDatabase.equals(computeWebApplication)) {
+                // means we can set the private address as reference the database endpoint
+                //TODO only set privateAddress or also publicAddress?
+                computeMysqlDatabase.setPrivateAddress(Fn.fnGetAtt(toAlphanumerical(mysqlDatabase.getEntityName()),
```

Comment:
Why is the private address set in both Visitor classes?

Reply:
I try to cover two possibilities here.
1. `PrepareModelNodeVisitor.java` sets the private address of the compute node to the database endpoint exactly when the database is the only node hosted on the compute. That means this compute node completely belongs to the database and represents it (by providing the endpoint)
2. `PrepareModelRelationshipVisitor.java` is used in the case of our current lamp-model where both the webapplication and the database are hosted on one compute. I only feel safe to set the private address of a compute node to reference the database if there is a connects to relationship between the two nodes sitting on top of the compute node.
Here I see the problem tackled in the previous meeting. What if someone references the webapplication as well by calling the private address of compute node it is hosted on (which is valid). He would get the database endpoint instead.

Reply:
I just had following alternative idea:
- when handling the `dbms` node;
  1. store the initial value of `compute#privateAddress`
  2. apply the AWS `Fn` fix (`compute.setPrivateAddress(Fn....)`)
  3. write all mysql dbms related configuration to aws template file
  4. restore value of `compute#privateAddress`
  
Could this work?

Reply:
I will look into this

Reply:
I don't think this will work. I actually need the privateAddress to be set to the database endpoint when I visit the webApplication node and not the dbms/database node.
I think the current solution is the best there is and will work for most cases.